router-bridge/src/api_schema.rs

```diff
@@ -0,0 +1,68 @@
+/*!
+# Generate an API schema from an sdl.
+*/
+
+use crate::error::Error;
+use crate::js::Js;
+use serde::{Deserialize, Serialize};
+use std::fmt::Display;
+use thiserror::Error;
+
+/// An error which occurred during JavaScript api schema generation.
+///
+/// The shape of this error is meant to mimick that of the error created within
+/// JavaScript, which is a [`GraphQLError`] from the [`graphql-js`] library.
+///
+/// [`graphql-js']: https://npm.im/graphql
+/// [`GraphQLError`]: https://github.com/graphql/graphql-js/blob/3869211/src/error/GraphQLError.js#L18-L75
+#[derive(Debug, Error, Serialize, Deserialize, PartialEq, Clone)]
+pub struct ApiSchemaError {
```

Comment:
doesn't have to be part of this PR but we use a similar error type across most of these packages. i wonder if we might consider writing some higher level type in `apollo-federation-types` where improvements would be delivered to all of the sub-packages. for instance, i imagine that there's some sort of case where this struct doesn't quite deserialize things properly (though I haven't investigated the root cause as of yet) https://github.com/apollographql/federation-rs/issues/42

Reply:
Agree, but let's keep refactor later. This may become redundant soon.